extensions/windows-event-log/ConsumeWindowsEventLog.cpp

```diff
@@ -572,6 +572,9 @@ bool ConsumeWindowsEventLog::createEventRender(EVT_HANDLE hEvent, EventRender& e
 
       for (const auto &mapEntry : walker.getIdentifiers()) {
         // replace the identifiers with their translated strings.
+        if (mapEntry.first.empty() || mapEntry.second.empty()) {
+          continue;  // This is most probably a result of a failed ID resolution
+        }
```

Comment:
I would move this check to the top of `StringUtils::replaceAll()`, as a `"" -> ""` replacement should always be a no-op, and there is no reason to potentially iterate over a long string character-by-character in these cases.